### ISSUE TITLE
pyo3-0.20.2 fails to build on systems without AtomicI64

### ISSUE THREAD

**Issue by he32** [PyO3/pyo3#4084]
### Bug Description

On a system which doesn't have 64-bit atomic types, pyo3 fails to build with

```
error[E0432]: unresolved import `std::sync::atomic::AtomicI64`
 --> /usr/pkgsrc/devel/py-pydantic-core/work/vendor/pyo3-0.20.2/src/impl_/pymodule.rs:6:25
  |
6 | use std::sync::atomic::{AtomicI64, Ordering};
  |                         ^^^^^^^^^
  |                         |
  |                         no `AtomicI64` in `sync::atomic`
  |                         help: a similar name exists in the module: `AtomicI32`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `pyo3` (lib) due to previous error
```

### Steps to Reproduce

1. Try to build e.g. pydantic-core (which includes this module) on an ILP32 CPU without 64-bit atomics
2. Watch it fail as per the above.

### Backtrace

```shell
See above, this is a build failure.
```

### Your operating system and version

NetBSD/10.0 macppc

### Your Python version (`python --version`)

Python 3.10.9

### Your Rust version (`rustc --version`)

rustc 1.77.1 (7cf61ebde 2024-03-27) (built from a source tarball)

### Your PyO3 version

0.20.2

### How did you install python? Did you use a virtualenv?

Built via source from NetBSD's pkgsrc system.

**Comment by davidhewitt** [comment]
Duplicate of #3614 

Fixed in 0.20.3, please update.

**Comment by he32** [comment]
Ah, thanks!  I've updated pkgsrc's py-pydantic-core to get a newer pyo3.
